inference-engine/src/cldnn_engine/CMakeLists.txt

```diff
@@ -22,19 +22,17 @@ ie_add_plugin(NAME ${TARGET_NAME}
               VERSION_DEFINES_FOR cldnn_engine.cpp)
 
 target_link_libraries(${TARGET_NAME} PRIVATE clDNN_lib pugixml
-                                             inference_engine inference_engine_legacy
-                                             inference_engine_transformations)
-
-if(USE_CNNNETWORK_LPT)
-    target_link_libraries(${TARGET_NAME} PRIVATE inference_engine_lp_transformations_legacy)
-else()
-    target_link_libraries(${TARGET_NAME} PRIVATE inference_engine_lp_transformations)
-endif()
+                                             inference_engine
+                                             inference_engine_transformations
+                                             inference_engine_lp_transformations)
 
 set(CLDNN_TOP_FOLDER "${IE_MAIN_SOURCE_DIR}/thirdparty/clDNN")
 target_include_directories(${TARGET_NAME} PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}
+        $<TARGET_PROPERTY:inference_engine_plugin_api,INTERFACE_INCLUDE_DIRECTORIES>
         $<TARGET_PROPERTY:inference_engine_transformations,INTERFACE_INCLUDE_DIRECTORIES>
+        $<TARGET_PROPERTY:ngraph::ngraph,INTERFACE_INCLUDE_DIRECTORIES>
+        ${IE_MAIN_SOURCE_DIR}/src/inference_engine
```

Comment:
Why do we need `${IE_MAIN_SOURCE_DIR}/src/inference_engine`? It's prohibited for plugins to use internals of IE. Only stuff from plugin API can be used.

Reply:
it's needed for `#include <cnn_network_ngraph_impl.hpp>` which, in its turn, is needed to do network clone without legacy lib. I'll remove it, once we have a function in the public plugin API to clone network.

Reply:
Let's continue in https://github.com/openvinotoolkit/openvino/pull/2506#discussion_r516113662